dot/core/service.go

```diff
@@ -153,12 +154,17 @@ func NewService(cfg *Config) (*Service, error) {
 			babeKill:         babeKill,
 			isBabeAuthority:  true,
 			lock:             chanLock,
-			closed:           false,
 			syncLock:         syncerLock,
 			blockNumOut:      cfg.SyncChan,
 			respOut:          respChan,
 		}
 
+		// thread safe way to change closed status
+		canLock := atomic.CompareAndSwapUint32(&srv.closed, 0, 1)
+		if !canLock {
+			panic("[core] Error when trying to change Service status from stopped to started.")
```

Comment:
it would be preferred not to panic here and return error instead

Reply:
sure